main.tf

```
@@ -1,16 +1,129 @@
+########################################
+resource "aws_cloudwatch_log_group" "this" {
+  name_prefix = local.cloudwatch_log_group_name_prefix
+  tags        = var.tags
+}
+
+########################################
+# LB
+########################################
+resource "aws_lb_target_group" "this" {
+  name_prefix = local.lb_target_group_name_prefix
+  port        = var.target_group_port
+  protocol    = "HTTP"
+  tags        = var.tags
+  target_type = "ip"
+  vpc_id      = var.vpc_id
+
+  health_check {
+    healthy_threshold   = var.health_check.healthy_threshold
+    interval            = var.health_check.interval
+    port                = var.health_check.port
+    protocol            = var.health_check.protocol
+    unhealthy_threshold = var.health_check.unhealthy_threshold
+  }
 
-  enforce_case = "UPPER"
-  names        = [var.name]
-  tags         = var.tags
+  lifecycle {
+    create_before_destroy = true
+  }
 }
 
-locals {
-  # tflint-ignore: terraform_unused_declarations
-  name = module.tags.name
-  # tflint-ignore: terraform_unused_declarations
-  tags = module.tags.tags_no_name
+resource "aws_lb_listener" "this" {
+  depends_on        = [aws_lb_target_group.this]
+  load_balancer_arn = var.load_balancer_arn
+  port              = var.listener_port
+  protocol          = "HTTP" #tfsec:ignore:AWS004
+
+  default_action {
+    target_group_arn = aws_lb_target_group.this.arn
+    type             = "forward"
+  }
+}
+
+########################################
+# ECS
+########################################
+module "container_definition" {
+  source  = "cloudposse/ecs-container-definition/aws"
+  version = "0.25.0"
+
+  environment       = var.environment_variables
+  container_cpu     = var.task_cpu
+  container_image   = var.container_image
+  container_memory  = var.task_memory
+  container_name    = local.container_name
+  log_configuration = local.log_configuration
+  port_mappings     = local.port_mappings
+  secrets           = var.secrets
+}
+
+resource "aws_ecs_task_definition" "this" {
+  container_definitions    = module.container_definition.json
+  cpu                      = var.task_cpu
+  execution_role_arn       = aws_iam_role.ecs_exec.arn
+  family                   = "${var.name}-task-def"
+  memory                   = var.task_memory
+  network_mode             = var.network_mode
+  requires_compatibilities = [var.launch_type]
+  tags                     = var.tags
+  task_role_arn            = aws_iam_role.ecs_task.arn
+}
+
+resource "aws_ecs_service" "this" {
+  cluster         = var.cluster_name
+  desired_count   = var.task_desired_count
+  launch_type     = var.launch_type
+  name            = "${var.name}-service"
+  task_definition = aws_ecs_task_definition.this.arn
+
+  load_balancer {
+    container_name   = local.container_name
+    container_port   = var.container_port
+    target_group_arn = aws_lb_target_group.this.arn
+  }
+
+  lifecycle {
+    # Subsequent deploys are done via code pipeline
```

```suggestion
    # Subsequent deploys are likely to be done through an external deployment pipeline
```
(might not be code pipeline)